lib/modules.nix

```
@@ -367,7 +367,14 @@ let
             if m._class == null || m._class == class
             then m
             else
-              throw "The module ${m._file or m.key} was imported into ${class} instead of ${m._class}."
+              throw ''
+                The module `${m._file or m.key}` (class: `${m._class}`) cannot be imported into a module evaluation that expects class `${class}`.
+
+                Help:
+                - Ensure that you are importing the correct module.
+                - Verify that the module's class (${m._class}) matches the expected class (${class}).
```

We could use more real syntax here perhaps.

```suggestion
                - Verify that the module's `_class`, ${lib.strings.escapeNixString m._class} matches the expected `class` ${lib.strings.escapeNixString class}.
```

I guess you could also do something like this on the first line, although it isn't wrong. Conceptually it is like an identifier - or at least that's how we've used it, but then I guess you could use `escapeNixIdentifier` which quotes less.